scipy-stubs/signal/_spectral_py.pyi

```diff
@@ -1,97 +1,209 @@
+@overload
+def spectrogram(
+    x: npt.NDArray[np.generic],
+    fs: AnyReal,
+    window: _GetWindowArgument | _ArrayLikeFloat_co,
+    nperseg: AnyInt | None,
+    noverlap: AnyInt | None,
+    nfft: AnyInt | None,
+    detrend: _Detrend,
+    return_onesided: op.CanBool,
+    scaling: _Scaling,
+    axis: op.CanIndex,
+    mode: Literal["complex"],
+) -> tuple[_Array_f8, _Array_f8, _ArrayComplex]: ...
+
+# complex mode (keyword)
+
+@overload
 def spectrogram(
-    x: Untyped,
-    fs: float = 1.0,
-    window: Untyped = ("tukey", 0.25),
-    nperseg: int | None = None,
-    noverlap: int | None = None,
-    nfft: int | None = None,
-    detrend: str | Literal[False] | UntypedCallable = "constant",
-    return_onesided: bool = True,
-    scaling: str = "density",
-    axis: int = -1,
-    mode: str = "psd",
-) -> Untyped: ...
-def check_COLA(window: Untyped, nperseg: int, noverlap: int, tol: float = 1e-10) -> Untyped: ...
-def check_NOLA(window: Untyped, nperseg: int, noverlap: int, tol: float = 1e-10) -> Untyped: ...
+    x: npt.NDArray[np.generic],
+    fs: AnyReal = 1.0,
+    window: _GetWindowArgument | _ArrayLikeFloat_co = ("tukey", 0.25),
+    nperseg: AnyInt | None = None,
+    noverlap: AnyInt | None = None,
+    nfft: AnyInt | None = None,
+    detrend: _Detrend = "constant",
+    return_onesided: op.CanBool = True,
+    scaling: _Scaling = "density",
+    axis: op.CanIndex = -1,
+    *,
+    mode: Literal["complex"],
+) -> tuple[_Array_f8, _Array_f8, _ArrayComplex]: ...
+
+#
+def check_COLA(
+    window: _GetWindowArgument | _ArrayLikeFloat_co,
+    nperseg: AnyInt,
+    noverlap: AnyInt,
+    tol: AnyReal = 1e-10,
+) -> bool: ...
+def check_NOLA(
+    window: _GetWindowArgument | _ArrayLikeFloat_co,
+    nperseg: AnyInt,
+    noverlap: AnyInt,
+    tol: AnyReal = 1e-10,
+) -> bool: ...
 def stft(
-    x: Untyped,
-    fs: float = 1.0,
-    window: str = "hann",
-    nperseg: int = 256,
-    noverlap: int | None = None,
-    nfft: int | None = None,
-    detrend: bool = False,
-    return_onesided: bool = True,
-    boundary: str = "zeros",
-    padded: bool = True,
-    axis: int = -1,
-    scaling: str = "spectrum",
-) -> Untyped: ...
+    x: _ArrayLikeComplex_co,
+    fs: AnyReal = 1.0,
+    window: _GetWindowArgument | _ArrayLikeFloat_co = "hann",
+    nperseg: AnyInt = 256,
+    noverlap: AnyInt | None = None,
+    nfft: AnyInt | None = None,
+    detrend: _Detrend = False,
+    return_onesided: op.CanBool = True,
+    boundary: _Boundary = "zeros",
+    padded: op.CanBool = True,
+    axis: op.CanIndex = -1,
+    scaling: _LegacyScaling = "spectrum",
+) -> tuple[_Array_f8, _Array_f8, _ArrayComplex]: ...
+
+# input_onesided is `True`
+@overload
+def istft(
+    Zxx: _ArrayLikeComplex_co,
+    fs: AnyReal = 1.0,
+    window: _GetWindowArgument | _ArrayLikeFloat_co = "hann",
+    nperseg: AnyInt | None = None,
+    noverlap: AnyInt | None = None,
+    nfft: AnyInt | None = None,
+    input_onesided: Literal[True] = True,
```

Comment:
I've seen quite a bit of internal scipy code that uses `1` for `True` and `0` for `False` (probably written C-developer 😋). 
So in case you want to allow that, you could include a `1` in this literal (and a `0` in the `False` literals below). But as far as I'm concerned, it's also fine to leave it like this, and just require users to do as the docs say they should 🤷🏻.